src/interfaces/ecpg/test/meson.build

```diff
@@ -0,0 +1,74 @@
+# TODO: ECPG is not enabled for windows yet
+if host_machine.system() == 'windows'
+  subdir_done()
+endif
+
+pg_regress_ecpg_sources = pg_regress_c + files(
+  'pg_regress_ecpg.c',
+)
+
+pg_regress_ecpg = executable('pg_regress_ecpg',
+  pg_regress_ecpg_sources,
+  c_args: pg_regress_cflags,
+  include_directories: [pg_regress_inc, include_directories('.')],
+  dependencies: [frontend_code],
+  kwargs: default_bin_args + {
+    'install': false
+  },
+)
+
+# create .c files and executables from .pgc files
+ecpg_test_exec_kw = {
+  'dependencies': [frontend_code, libpq],
+  'include_directories': [ecpg_inc],
+  'link_with': [ecpglib, ecpg_compat, ecpg_pgtypes],
+  'build_by_default': false,
+}
+
+ecpg_preproc_test_command_start = [
+  ecpg_exe, 
+  '--regression',
+  '-I@CURRENT_SOURCE_DIR@',
+  '-I@SOURCE_ROOT@' + '/src/interfaces/ecpg/include/',
+]
+ecpg_preproc_test_command_end = [
+  '-o', '@OUTPUT@', '@INPUT@'
+]
+
+ecpg_test_dependencies = []
+
+subdir('connect')
+subdir('sql')
+subdir('pgtypeslib')
+subdir('preproc')
+subdir('compat_informix')
+subdir('compat_oracle')
+subdir('thread')
+
+ecpg_test_files = files(
+  'ecpg_schedule',
+)
+
+ecpg_regress_args = [
+  '--dbname=ecpg1_regression,ecpg2_regression',
+  '--create-role=regress_ecpg_user1,regress_ecpg_user2',
+  '--encoding=SQL_ASCII',
+]
+
+tests += {
+  'name': 'ecpg',
+  'sd': meson.current_build_dir(),
+  'bd': meson.current_build_dir(),
+  'regress': {
+    'test_runner': pg_regress_ecpg,
+    'expecteddir': meson.current_source_dir(),
```

Comment:
Maybe I'm just slow right now. But how does this differ from --inputdir? That's also set to the source directory? Oh, I see - you intentionally set 'sd' to something other than the source directory? I don't think that's good...

Reply:
Yes, it was intentional. But since generated .c files and executables(in that case, these are input files) are in the build directory, I set sd as a build directory. Other way could be generating .c files and executables in the source directory but that also seems wrong to me. Do you have any other idea?

Current logic is:
.pgc files are in the source directory (we are getting these as an input at custom_target())
expected files are in the build directory (expecteddir is added)
.c and executables in the build directory (I set sd as a build directory)

Reply:
There can be small confusion, source files are not generated by setting the source directory to the build directory. Source files of ecpg_test are taken from build directory.

There are two steps:
1. Generating .c and executables, which is done by setting custom_target and executables. We are getting these .pgc files from source directory; then create .c and executables in the build directory.
2. These executables and .c files are used in ecpg regression test. ecpg_test needs to read these files and since these .c and executables are in the build directory, I set ecpg_test's sd as a build directory.

Reply:
> There can be small confusion, source files are not generated by setting the source directory to the build directory. Source files of ecpg_test are taken from build directory.

My point is that you're setting `'sd'` to `meson.current_build_dir()`, but `'sd'` stands for source directory. If you need to change where the test gets the input from or such, you should override that parameter, instead of "lying" about 'sd'.

Reply:
Updated.